File: src/main/java/at/hrechny/predictionsbot/util/TimeZoneUtils.java
package at.hrechny.predictionsbot.util;

import us.dustinj.timezonemap.TimeZoneMap;

public class TimeZoneUtils {

  private final static TimeZoneMap timeZoneMap = TimeZoneMap.forEverywhere();

  public static String getTimeZone(Float latitude, Float longitude) {
    var timeZone = timeZoneMap.getOverlappingTimeZone(latitude, longitude);
    return timeZone != null ? timeZone.getZoneId() : null;
  }

}

File: src/main/kotlin/at/hrechny/predictionsbot/util/TimeZoneUtils.kt
package at.hrechny.predictionsbot.util

import us.dustinj.timezonemap.TimeZoneMap

object TimeZoneUtils {
    private val timeZoneMap: TimeZoneMap = TimeZoneMap.forEverywhere()

    @JvmStatic
    fun getTimeZone(latitude: Float, longitude: Float): String? {
        val timeZone = timeZoneMap.getOverlappingTimeZone(latitude.toDouble(), longitude.toDouble())
        return timeZone?.zoneId
    }
}
